File: translator.js
#! /usr/bin/env node
const fs = require('fs');
const translate = require('translation-google');
// const translate = require('translate-google');

let writeTranslations = (localePath, data) => {
  fs.writeFile(localePath, JSON.stringify(data), err => {
    if (err) throw err;
  });
};

let generateTranslation = () => {
  let locales = fs.readdirSync('./src/locales');

  locales.map(locale => {
    if (locale !== '_build' && locale !== 'en') {
      let localePath = './src/locales/' + locale + '/' + 'messages.json';
      let data = JSON.parse(fs.readFileSync(localePath));
      let translatedData = data;
      let keys = Object.keys(data);

      keys.forEach((text, index) => {
        translate(text, { from: 'en', to: locale })
          .then(res => {
            translatedData[text] = res.text;

            // if (res.text.length === 0) {
            // }
            if (index === keys.length - 1) {
              writeTranslations(localePath, translatedData);
            }
          })
          .catch(_err => {});
      });
    }
  });
};

generateTranslation();

File: translator.py
#!/usr/bin/env python3

import os
import sys
import json
from googletrans import Translator


def translate(project_path):

    print("Path", project_path)
    if os.path.exists(project_path):

        if not project_path.endswith('/'):
            project_path += '/'

        locales_path = '{0}src/locales'.format(project_path)
        locales = os.listdir(locales_path)
        locales.remove('_build')
        locales.remove('en')
        translator = Translator()

        for locale in locales:
            target_language = locale
            locale_path = os.path.join(locales_path, locale, 'messages.json')
            new_data = dict()

            with open(locale_path, 'r') as f:
                data = json.load(f)
                results = translator.translate(
                    list(set(data.keys())),
                    dest=target_language
                )

                for item in results:
                    new_data.update({item.origin: item.text})

            with open(locale_path, 'w', encoding='utf-8') as f:
                json.dump(new_data, f, ensure_ascii=False, indent=4)

    else:
        print('Project path is invalid')


if __name__ == "__main__":
    if len(sys.argv):
        path = sys.argv[1:][0]
        translate(path)
